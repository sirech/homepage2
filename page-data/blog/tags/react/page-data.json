{
    "componentChunkName": "component---src-templates-blog-index-js",
    "path": "/blog/tags/react",
    "result": {"pageContext":{"siteMetadata":{"title":"Mario Fernandez","description":"This is the blog from Mario Fernandez, a Software Developer working for ThoughtWorks","author":"Mario Fernandez","url":"https://hceris.com","twitter":"@sirech"},"group":[{"frontmatter":{"layout":"post","title":"Applications Instead of Libraries","path":"/applications-instead-of-libraries/","categories":["Architecture","React","Micro frontends","JavaScript"],"date":"2021/11/20","draft":false,"description":"Using module federation to implement micro frontends"}},{"frontmatter":{"layout":"post","title":"From Redux to the Context API: A Practical Migration Guide","path":"/redux-to-context-practical-migration-guide/","categories":["React","Redux","Context API","Architecture","State Management"],"date":"2021/06/14","draft":false,"description":"Let's migrate a React application built with Redux to the Context API and see how it looks like"}},{"frontmatter":{"layout":"post","title":"Starting you new React application with Redux? Consider a simpler approach using the Context API","path":"/starting-a-new-app-with-redux-consider-context-api-first/","categories":["React","Redux","Context API","Architecture"],"date":"2021/05/11","draft":false,"description":"Traditionally, starting a new React application means adding Redux for state management. Let's talk about the Context API as an alternative"}},{"frontmatter":{"layout":"post","title":"Authorization Code Flow with PKCE (OAuth) in a React application","path":"/oauth-authorization-code-flow-pkce-for-react/","categories":["Auth0","OAuth","React"],"date":"2021/03/29","draft":false,"description":"SPAs implementing OAuth should pick Authorization Code Flow with PKCE for maximum security. Let's implement it in React with help of Auth0"}},{"frontmatter":{"layout":"post","title":"A directory structure for React projects","path":"/a-directory-structure-for-react-projects/","categories":["React","TypeScript","Code Organization"],"date":"2020/07/19","draft":false,"description":"How do you structure your components in a React application? I like to group mine around features close to the domain enforced by convention"}},{"frontmatter":{"layout":"post","title":"Fail a test in Jest if an unexpected network request happens","path":"/jest-fail-test-if-unexpected-network-request-happens/","categories":["React","Jest","Testing","react-testing-library","create-react-app","axios"],"date":"2019/12/08","draft":false,"description":"Automatically failing unit tests with Jest in React when network requests are made will make your tests more reliable and easier to maintain"}},{"frontmatter":{"layout":"post","title":"Custom Components in Formik","path":"/custom-components-in-formik/","categories":["React","Formik","react-hooks"],"date":"2019/08/15","draft":false,"description":"Formik is the new cool kid on the block for forms in react. Here is how to build custom input elements on top"}},{"frontmatter":{"layout":"post","title":"Migrating to Gatsby 2","path":"/migrating-to-gatsby2/","categories":["React","Gatsby"],"date":"2018/10/03","draft":false,"description":"Gatsby 2 is out! How did the migration go?"}},{"frontmatter":{"layout":"post","title":"Angular from the perspective of a React fan - Part 3","path":"/angular-from-react-part3/","categories":["JavaScript","React","Angular","TypeScript","angular-cli","redux"],"date":"2018/07/07","draft":false,"description":"Third part of my series on Angular, coming from a React background. This article focus on state management"}},{"frontmatter":{"layout":"post","title":"Angular from the perspective of a React fan - Part 2","path":"/angular-from-react-part2/","categories":["JavaScript","React","Angular","TypeScript","angular-cli"],"date":"2018/04/29","draft":false,"description":"Second part of my series on Angular, coming from a React background. This article focus on testing"}},{"frontmatter":{"layout":"post","title":"Angular from the perspective of a React fan - Part 1","path":"/angular-from-react-part1/","categories":["JavaScript","React","Angular","TypeScript","angular-cli"],"date":"2018/03/25","draft":false,"description":"First part of my series on Angular, coming from a React background. This article is about bootstrapping and TypeScript"}},{"frontmatter":{"layout":"post","title":"Migrating to Gatsby","path":"/migrating-to-gatsby/","categories":["React","Gatsby"],"date":"2018/03/18","draft":false,"description":"It all starts here. After using Create React App for a while, I decided to switch to Gatsby for my personal website"}}],"pathPrefix":"blog/tags/react","first":true,"last":true,"index":1,"pageCount":1,"additionalContext":{"siteMetadata":{"title":"Mario Fernandez","description":"This is the blog from Mario Fernandez, a Software Developer working for ThoughtWorks","author":"Mario Fernandez","url":"https://hceris.com","twitter":"@sirech"}}}},
    "staticQueryHashes": []}