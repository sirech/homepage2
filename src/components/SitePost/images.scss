@import '../../layouts/colors';
@import '../../layouts/fonts';

.figure {
  width: 100%;

  @media only screen and (min-width: 500px) {
    &--left,
    &--right {
      margin-left: auto;
      margin-right: auto;
      max-width: 315px;
    }

    &--left {
      float: left;
      margin-right: 1rem;
    }

    &--right {
      float: right;
      margin-left: 1rem;
    }
  }

  img {
    border: 1px solid $gray-700;
  }

  &--no-border img {
    border: 0;
  }

  &__image-wrap {
    border: 1px solid $gray-700;
    margin-bottom: 0;
  }

  &__image-link {
    border: 0;
  }

  &__caption {
    border-bottom: 1px solid $gray-700;
    font-size: $font-size-xs;
    margin-top: 0.5rem;
    padding: 0 0 0.5rem;

    blockquote & {
      border-bottom: 0;
    }

    &:empty {
      display: none;
    }
  }

  &__attribution {
    display: block;
    letter-spacing: 0.05rem;
    text-transform: uppercase;
  }
}
