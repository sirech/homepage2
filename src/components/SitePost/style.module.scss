@import '../../layouts/colors';
@import '../../layouts/fonts';

.articles {
  padding-top: 2rem;
}

.article {
  padding-bottom: 0.5rem;

  header {
    border-bottom: 1px solid $gray-200;
    margin-bottom: 1rem;
    padding-bottom: 0.5rem;
    word-break: break-word;

    h1 {
      color: $gray-600;
    }

    time {
      color: $gray-500;
      display: inline;
      font-size: $font-size-xs;
      margin-right: 0.5rem;
    }
  }
}

.pageContent {
  padding-bottom: 0.5rem;

  h2 {
    border-bottom: 3px solid $gray-400;
    border-top: 3px solid $gray-400;

    margin-top: 2rem;
    padding: 0.5rem 0;
  }

  h3 {
    border-bottom: 2px solid $gray-400;
    margin-top: 1.5rem;
    padding-bottom: 0.5rem;
  }

  h4 {
    border-bottom: 1px solid $gray-400;
    padding-bottom: 0.5rem;
  }

  p {
    margin: 1rem 0;
  }

  blockquote {
    border-left: 10px solid $gray-800;
    box-shadow: rgba(0, 0, 0, 0.2) 0 0 5px 0;
    font-style: italic;
    padding: 0.25rem 1.5rem;
  }
}

:global(.gatsby-highlight) {
  code {
    font-size: 0.8rem;
    white-space: pre-wrap;
  }
}

:global(.guide) {
  background-color: $white;
  border: 1px solid map-get($theme-colors, 'light');
  border-radius: 4px;
  box-shadow: 0 2px 1px -1px rgba(0, 0, 0, 0.2), 0 1px 1px 0 rgba(0, 0, 0, 0.14),
    0 1px 3px 0 rgba(0, 0, 0, 0.12);

  margin-bottom: 1rem;
  padding: 1rem;

  transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;

  ul {
    list-style: none;
    margin-bottom: 0;
  }

  h3 {
    margin-top: 0;
  }
}
