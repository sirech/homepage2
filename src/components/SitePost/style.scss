@import '../../layouts/colors.scss';
@import '../../layouts/fonts.scss';

.articles {
  padding-top: 2rem;

  article {
    padding-bottom: 0.5rem;

    header {
      padding-bottom: 0.5rem;
      margin-bottom: 1rem;
      border-bottom: 1px solid $gray-200;
      word-break: break-word;

      h1 {
        color: $gray-600;
      }

      .badge {
        margin-right: 0.2rem;
      }

      time {
        font-size: $font-size-xs;
        color: $gray-500;
        display: inline;
        margin-right: 0.5rem;
      }
    }

    .page-content {
      padding-bottom: 0.5rem;

      h2 {
        padding-top: 0.5rem;
        margin-bottom: 1rem;
      }

      p {
        margin: 1rem 0;
      }

      blockquote {
        background-color: $gray-100;
        border-left: 5px solid $gray-200;
        padding: 0.25rem 1.5rem;

        a {
          display: block;
          word-break: break-word;
        }
      }
    }
  }

  .gatsby-highlight {
    code {
      font-size: 0.8rem;
      white-space: pre-wrap;
    }
  }

  .guide {
    background-color: $gray-100;
    border: 1px solid $gray-300;
    border-radius: 15px;

    padding: 1rem;

    ul {
      margin-bottom: 0;
    }
  }
}
