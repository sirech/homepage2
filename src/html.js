import React from 'react'
import PropTypes from 'prop-types'
import Helmet from 'react-helmet'

let stylesStr
if (process.env.NODE_ENV === 'production') {
  try {
    // eslint-disable-next-line
    stylesStr = require('!raw-loader!../public/styles.css')
  } catch (e) {
    console.log(e)
  }
}

class HTML extends React.Component {
  render() {
    const head = Helmet.rewind()
    let css
    if (process.env.NODE_ENV === 'production') {
      css = (
        <style
          id="gatsby-inlined-css"
          dangerouslySetInnerHTML={{ __html: stylesStr }}
        />
      )
    }

    return (
      <html lang="en">
        <head>
          <meta charSet="utf-8" />
          <meta httpEquiv="X-UA-Compatible" content="IE=edge" />
          <meta
            name="viewport"
            content="width=device-width, initial-scale=1.0"
          />
          {this.props.headComponents}
          {css}
          <link
            href="/img/apple-touch-icon.png"
            rel="apple-touch-icon"
            sizes="180x180"
          />
          <link href="/img/favicon.ico" rel="icon" type="image/x-icon" />
        </head>
        <body itemScope itemType="http://schema.org/WebPage">
          <div
            id="___gatsby"
            dangerouslySetInnerHTML={{ __html: this.props.body }}
          />
          {this.props.postBodyComponents}
        </body>
      </html>
    )
  }
}

HTML.propTypes = {
  body: PropTypes.string.isRequired,
  headComponents: PropTypes.node,
  postBodyComponents: PropTypes.node,
}

export default HTML
