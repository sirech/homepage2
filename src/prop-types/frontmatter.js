import PropTypes from 'prop-types'

export default PropTypes.shape({
  categories: PropTypes.arrayOf(PropTypes.string).isRequired,
  date: PropTypes.string.isRequired,
  layout: PropTypes.string.isRequired,
  path: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  draft: PropTypes.bool.isRequired,
  description: PropTypes.string,
}).isRequired
