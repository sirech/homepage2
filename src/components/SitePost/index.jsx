import React from 'react'
import PropTypes from 'prop-types'
import cx from 'classnames'

import Link from 'gatsby-link'
import Container from '../Container'
import Categories from '../Categories'
import Share from '../Share'

import styles from './style.module.scss'
import './images.scss'

import frontmatterType from '../../prop-types/frontmatter'
import siteType from '../../prop-types/site'

class SitePost extends React.Component {
  render() {
    const { data, site } = this.props
    const { frontmatter, html } = data
    const { title, path, date, categories } = frontmatter

    return (
      <Container>
        <div className={styles.articles}>
          <article key={path} className={cx(`${styles.article}`)}>
            <header>
              <Link style={{ boxShadow: 'none' }} to={path}>
                <h1>{title}</h1>
                <time dateTime={date}>{date}</time>
              </Link>
              <Categories categories={categories} />
            </header>
            <section
              className={cx(`${styles.pageContent}`, 'clearfix')}
              dangerouslySetInnerHTML={{
                __html: html,
              }}
            />

            <section>
              <Share site={site} path={path} />
            </section>
          </article>
        </div>
      </Container>
    )
  }
}

SitePost.propTypes = {
  data: PropTypes.shape({
    frontmatter: frontmatterType,
    html: PropTypes.string.isRequired,
  }).isRequired,
  site: siteType,
}

export default SitePost
