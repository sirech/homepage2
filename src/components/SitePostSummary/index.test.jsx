import React from 'react'
import { render } from '@testing-library/react'
import SitePost from './index'

import frontmatter from '../../fixtures/frontmatter'
import html from '../../fixtures/html'

describe('components', () => {
  describe('SitePostSummary', () => {
    it('renders correctly', () => {
      const data = { html: html(), frontmatter: frontmatter() }
      const { asFragment } = render(<SitePost data={data} />)
      expect(asFragment()).toMatchSnapshot()
    })
  })
})
