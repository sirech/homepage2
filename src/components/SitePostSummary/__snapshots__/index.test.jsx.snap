// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components SitePostSummary renders correctly 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <div
      class="articles"
    >
      <article
        class="article articleBox"
      >
        <header>
          <a
            href="/the-post"
          >
            <h3>
              My Post
            </h3>
          </a>
          <time
            datetime="2018/03/01"
          >
            2018/03/01
          </time>
          <a
            href="/blog/tags/javascript"
          >
            <span
              class="tag badge mr-1"
            >
              JavaScript
            </span>
          </a>
          <a
            href="/blog/tags/react"
          >
            <span
              class="tag badge mr-1"
            >
              React
            </span>
          </a>
        </header>
        <section>
          this is the description
        </section>
      </article>
    </div>
  </div>
</DocumentFragment>
`;
